backend/graphql/resolvers/classResolvers.ts

```diff
@@ -67,6 +66,14 @@ const classResolvers = {
       return classService.createStudent(student, classId);
     },
   },
+  ClassResponseDTO: {
+    teacher: async (parent: ClassResponseDTO) => {
+      return userService.getUserById(parent.teacher);
+    },
+    testSessions: async (parent: ClassResponseDTO) => {
+      return testSessionService.getTestSessionsByClassId(parent.id);
+    },
+  },
```

Comment:
wait, i'm not entirely sure what's going on here. does this overload the existing imported `ClassResponseDTO`? or how does it work

Reply:
Yeah, exactly. So the way I set it up is that the main resolver returns a DTO with just the ID for this field. And then there are two cases: if the frontend requested the `testSessions` field, then Apollo will additionally run this resolver, and the return value replaces the original ID; otherwise, the ID is just discarded. So the net effect is that we only hit the database an extra time if the field is actually requested, which seems to shave a TON of time off the response time.